document_merge_service/api/authentication.py

```diff
@@ -0,0 +1,109 @@
+import requests
+from django.conf import settings
+from django.core.cache import cache
+from django.utils.encoding import smart_text
+from django.utils.translation import ugettext as _
+from jose import ExpiredSignatureError, JWTError, jwt
+from rest_framework import authentication, exceptions
+
+
+class AnonymousUser(object):
+    def __init__(self):
+        self.username = None
```

Comment:
If you set `self.username` to "AnonymousUser", you won't need to overload `__str__()` in the `OIDCUser` class later on.

Reply:
that's not desired as self.username is potentially stored to database and no user is represented as None.